DataFormats/Detectors/GlobalTracking/include/DataFormatsGlobalTracking/RecoContainer.h

```diff
@@ -580,6 +595,11 @@ struct RecoContainer {
   auto getTOFClusters() const { return getSpan<o2::tof::Cluster>(GTrackID::TOF, CLUSTERS); }
   auto getTOFClustersMCLabels() const { return mcTOFClusters.get(); }
 
+  // HMPID clusters
+  auto getHMPClusters() const { return getSpan<o2::hmpid::Cluster>(GTrackID::HMP, CLUSTERS); }
+  auto getHMPTriggers() const { return getSpan<o2::hmpid::Trigger>(GTrackID::HMP, CLUSREFS); }
```

Comment:
Your matches are also grouped per trigger, right? Then there should be separate 
```
auto getHMPClusterTriggers() const { return getSpan<o2::hmpid::Trigger>(GTrackID::HMP, CLUSREFS); }
auto getHMPClusters() const { return getSpan<o2::hmpid::Cluster>(GTrackID::HMP, CLUSTERS); }
```
and e.g.
```
auto getHMPMatchTriggers() const { return getSpan<o2::hmpid::Trigger>(GTrackID::HMP, TRACKREFS); }
auto getHMPMatches() const { return getSpan<o2::hmpid::MatchInfoHMP>(GTrackID::HMP, MATCHES); }
```

Reply:
ok, and  I have to keep this then?

const o2::dataformats::MatchInfoHMP& getHMPMatch(GTrackID id) const { return getObject<o2::dataformats::MatchInfoHMP>(id, MATCHES); }

Reply:
As you prefer, the same can be obtained as 
```
auto mtspan = recoCont.getHMPMatches();
auto& match = mtspan[id.getIndex()];
```
Note that each `getHMPMatch(GTrackID id) ` would go via new span creation. It is not very expensive but still more expensive than getting span once then iterating over its elements.